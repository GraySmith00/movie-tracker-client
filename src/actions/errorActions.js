export const setRegisterErrorState = errorMessage => ({
  type: 'SET_REGISTER_ERROR_STATE',
  errorMessage
});

export const setLoginErrorState = errorMessage => ({
  type: 'SET_LOGIN_ERROR_STATE',
  errorMessage
});

export const setFavoritesErrorState = errorMessage => ({
  type: 'SET_FAVORITES_ERROR_STATE',
  errorMessage
});
