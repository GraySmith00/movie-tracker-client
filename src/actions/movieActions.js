export const addNowPlaying = movies => ({
  type: 'ADD_NOW_PLAYING',
  movies
});

export const toggleMovieStatus = changedMovie => ({
  type: 'TOGGLE_MOVIE_STATUS',
  changedMovie
});

export const addFavoriteToState = movieId => ({
  type: 'ADD_FAVORITE_TO_STATE',
  movieId
});

export const removeFavoriteFromState = movieId => ({
  type: 'REMOVE_FAVORITE_FROM_STATE',
  movieId
});

export const populateFavoritesState = movieIds => ({
  type: 'POPULATE_FAVORITES_STATE',
  movieIds
});

export const clearFavorites = () => ({
  type: 'CLEAR_FAVORITES'
});
