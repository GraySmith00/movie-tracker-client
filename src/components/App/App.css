.app {
  display: flex; }

.header-container {
  width: 14rem;
  height: 100vh;
  background: #00b4db;
  /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #0083b0, #00b4db);
  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right, #0083b0, #00b4db);
  /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */ }

.main-container {
  flex: 10;
  height: 100vh;
  overflow: scroll; }
