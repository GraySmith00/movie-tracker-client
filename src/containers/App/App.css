.app {
  display: flex; }

.header-container {
  width: 14rem;
  height: 100vh;
  background: #00b4db;
  /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #0083b0, #00b4db);
  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right, #0083b0, #00b4db);
  /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */ }

.main-container {
  flex: 10;
  height: 100vh;
  overflow: scroll; }

.modal-trailer {
  border: none;
  height: 500px;
  width: 50vw; }

.modal {
  z-index: 1;
  position: absolute;
  top: 10rem;
  left: 35rem;
  border: 10px solid rgba(113, 137, 137, 0.5);
  overflow: auto;
  display: block;
  border-radius: 5px; }

.remove-modal {
  position: absolute;
  left: 1rem;
  top: 1rem;
  cursor: pointer;
  color: red;
  background-color: #f9b6b6;
  border-radius: 5px; }

@media screen and (max-width: 1550px) {
  .modal {
    position: absolute;
    top: 10rem;
    left: 25rem; }
  .modal-trailer {
    border: none;
    width: 60vw;
    height: 450px; } }

@media screen and (max-width: 1300px) {
  .modal {
    position: absolute;
    top: 10rem;
    left: 20rem; }
  .modal-trailer {
    border: none;
    width: 60vw;
    height: 400px; } }

@media screen and (max-width: 1050px) {
  .modal {
    position: absolute;
    top: 13rem;
    left: 17rem; }
  .modal-trailer {
    border: none;
    width: 61vw;
    height: 300px; } }

@media screen and (max-width: 780px) {
  .header-container {
    width: 100%;
    height: 12rem; }
  .app {
    flex-direction: column; }
  .modal-trailer {
    border: none;
    width: 73vw;
    height: 350px; }
  .modal {
    position: absolute;
    top: 20rem;
    left: 6rem; } }

@media screen and (max-width: 680px) {
  .modal {
    position: absolute;
    top: 20rem;
    left: 5rem; }
  .modal-trailer {
    border: none;
    width: 70vw;
    height: 350px; } }
