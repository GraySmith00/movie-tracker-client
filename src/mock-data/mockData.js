const nowPlayingResultsArray = {
  results: [
    {
      vote_count: 507,
      id: 345940,
      video: false,
      vote_average: 6.2,
      title: "The Meg",
      popularity: 173.315,
      poster_path: "/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg",
      original_language: "en",
      original_title: "The Meg",
      genre_ids: [28, 878, 53, 27],
      backdrop_path: "/ibKeXahq4JD63z6uWQphqoJLvNw.jpg",
      adult: false,
      overview:
        "A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.",
      release_date: "2018-08-09"
    },
    {
      vote_count: 1095,
      id: 353081,
      video: false,
      vote_average: 7.4,
      title: "Mission: Impossible - Fallout",
      popularity: 138.478,
      poster_path: "/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
      original_language: "en",
      original_title: "Mission: Impossible - Fallout",
      genre_ids: [12, 28, 53],
      backdrop_path: "/5qxePyMYDisLe8rJiBYX8HKEyv2.jpg",
      adult: false,
      overview:
        "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
      release_date: "2018-07-25"
    },
    {
      vote_count: 2833,
      id: 351286,
      video: false,
      vote_average: 6.6,
      title: "Jurassic World: Fallen Kingdom",
      popularity: 102.979,
      poster_path: "/c9XxwwhPHdaImA2f1WEfEsbhaFB.jpg",
      original_language: "en",
      original_title: "Jurassic World: Fallen Kingdom",
      genre_ids: [28, 12, 878],
      backdrop_path: "/3s9O5af2xWKWR5JzP2iJZpZeQQg.jpg",
      adult: false,
      overview:
        "Several years after the demise of Jurassic World, a volcanic eruption threatens the remaining dinosaurs on the island of Isla Nublar. Claire Dearing, the former park manager and founder of the Dinosaur Protection Group, recruits Owen Grady to help prevent the extinction of the dinosaurs once again.",
      release_date: "2018-06-06"
    }
  ]
};

export const modifiedNowPlaying = [
  {
    id: 345940,
    title: "The Meg",
    releaseDates: "2018-08-09",
    overview:
      "A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.",
    poster_path: "/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg"
  },
  {
    id: 353081,
    title: "Mission: Impossible - Fallout",
    img: "/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
    overview:
      "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
    releaseDates: "2018-07-25"
  },
  {
    id: 351286,
    title: "Jurassic World: Fallen Kingdom",
    img: "http://image.tmdb.org/t/p/original/c9XxwwhPHdaImA2f1WEfEsbhaFB.jpg",
    realeaseDates: "2018-06-06",
    overview:
      "Several years after the demise of Jurassic World, â€¦event the extinction of the dinosaurs on the island of Isla Nublar. Claire Dearing, the former park manager and founder of the Dinosaur Protection Group, recruits Owen Grady to help prevent the extinction of the dinosaurs once again."
  }
];
