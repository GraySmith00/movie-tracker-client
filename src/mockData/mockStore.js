export const mockStore = {
  movies: {
    nowPlaying: [
      {
        movie_id: 345940,
        title: 'The Meg',
        release_date: '2018-08-09',
        overview:
          'A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.',
        vote_average: 6.2,
        poster_path:
          'http://image.tmdb.org/t/p/original/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 353081,
        title: 'Mission: Impossible - Fallout',
        release_date: '2018-07-25',
        overview:
          'When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.',
        vote_average: 7.3,
        poster_path:
          'http://image.tmdb.org/t/p/original/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 402900,
        title: "Ocean's Eight",
        release_date: '2018-06-07',
        overview:
          "Debbie Ocean, a criminal mastermind, gathers a crew of female thieves to pull off the heist of the century at New York's annual Met Gala.",
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/MvYpKlpFukTivnlBhizGbkAe3v.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 507569,
        title: 'The Seven Deadly Sins: Prisoners of the Sky',
        release_date: '2018-08-18',
        overview:
          'The Seven Deadly Sins travel to a remote land in search of the phantom ingredient "sky fish." Meliodas and Hawk end up at a "Sky Palace" that exists above the clouds, where all the residents have wings. Meliodas is mistaken for a boy who committed a crime and is thrown in prison. Meanwhile, the residents are preparing a ceremony for defense against a ferocious beast that awakens once every 3,000 years. But the Six Knights of Black, a Demon Clan army, arrives and removes the seal on the beast in order to threaten the lives of the residents of Sky Palace. Meliodas and his allies meet the Six Knights of Black in battle.',
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/fK6zf4eN5xxkCO3MP9rDNWOq6Ys.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 493922,
        title: 'Hereditary',
        release_date: '2018-06-04',
        overview:
          "When Ellen, the matriarch of the Graham family, passes away, her daughter's family begins to unravel cryptic and increasingly terrifying secrets about their ancestry. The more they discover, the more they find themselves trying to outrun the sinister fate they seem to have inherited.",
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/4GFPuL14eXi66V96xBWY73Y9PfR.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 363088,
        title: 'Ant-Man and the Wasp',
        release_date: '2018-07-04',
        overview:
          "As Scott Lang awaits expiration of his term of house detention, Hope van Dyne and Dr. Hank Pym involve him in a scheme to rescue Mrs. van Dyne from the micro-universe into which she has fallen, while two groups of schemers converge on them with intentions of stealing Dr. Pym's inventions.",
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/rv1AWImgx386ULjcf62VYaW8zSt.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 399360,
        title: 'Alpha',
        release_date: '2018-08-17',
        overview:
          'After a hunting expedition goes awry, a young caveman struggles against the elements to find his way home.',
        vote_average: 5.7,
        poster_path:
          'http://image.tmdb.org/t/p/original/afdZAIcAQscziqVtsEoh2PwsYTW.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 455980,
        title: 'Tag',
        release_date: '2018-05-30',
        overview:
          'For one month every year, five highly competitive friends hit the ground running in a no-holds-barred game of tag they’ve been playing since the first grade. This year, the game coincides with the wedding of their only undefeated player, which should finally make him an easy target. But he knows they’re coming...and he’s ready.',
        vote_average: 6.9,
        poster_path:
          'http://image.tmdb.org/t/p/original/eXXpuW2xaq5Aen9N5prFlARVIvr.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 500664,
        title: 'Upgrade',
        release_date: '2018-06-01',
        overview:
          'A brutal mugging leaves Grey Trace paralyzed in the hospital and his beloved wife dead. A billionaire inventor soon offers Trace a cure — an artificial intelligence implant called STEM that will enhance his body. Now able to walk, Grey finds that he also has superhuman strength and agility — skills he uses to seek revenge against the thugs who destroyed his life.',
        vote_average: 7.4,
        poster_path:
          'http://image.tmdb.org/t/p/original/woLGROUdPWdBnLVFKYdGD3kFkPY.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 429300,
        title: 'Adrift',
        release_date: '2018-05-31',
        overview:
          "A true story of survival, as a young couple's chance encounter leads them first to love, and then on the adventure of a lifetime as they face one of the most catastrophic hurricanes in recorded history.",
        vote_average: 6.4,
        poster_path:
          'http://image.tmdb.org/t/p/original/5gLDeADaETvwQlQow5szlyuhLbj.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 260513,
        title: 'Incredibles 2',
        release_date: '2018-06-14',
        overview:
          'Elastigirl springs into action to save the day, while Mr. Incredible faces his greatest challenge yet – taking care of the problems of his three children.',
        vote_average: 7.7,
        poster_path:
          'http://image.tmdb.org/t/p/original/x1txcDXkcM65gl7w20PwYSxAYah.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 447200,
        title: 'Skyscraper',
        release_date: '2018-07-11',
        overview:
          "Framed and on the run, a former FBI agent must save his family from a blazing fire in the world's tallest building.",
        vote_average: 6,
        poster_path:
          'http://image.tmdb.org/t/p/original/5LYSsOPzuP13201qSzMjNxi8FxN.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 345887,
        title: 'The Equalizer 2',
        release_date: '2018-07-19',
        overview:
          "Robert McCall returns to deliver his special brand of vigilante justice -- but how far will he go when it's someone he loves?",
        vote_average: 6.3,
        poster_path:
          'http://image.tmdb.org/t/p/original/cQvc9N6JiMVKqol3wcYrGshsIdZ.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 476292,
        title: 'Maquia: When the Promised Flower Blooms',
        release_date: '2018-02-24',
        overview:
          "A story of encounters and partings interwoven between people; this is a human drama with feelings that touch one's heart gradually, which everyone has experienced at least once.",
        vote_average: 5.1,
        poster_path:
          'http://image.tmdb.org/t/p/original/qcnOKCPleLOTWPPgYI0YT1MOQwR.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 442249,
        title: 'The First Purge',
        release_date: '2018-07-04',
        overview:
          'To push the crime rate below one percent for the rest of the year, the New Founding Fathers of America test a sociological theory that vents aggression for one night in one isolated community. But when the violence of oppressors meets the rage of the others, the contagion will explode from the trial-city borders and spread across the nation.',
        vote_average: 6,
        poster_path:
          'http://image.tmdb.org/t/p/original/2slvblTroiT1lY9bYLK7Amigo1k.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 487558,
        title: 'BlacKkKlansman',
        release_date: '2018-08-09',
        overview:
          'Ron Stallworth, an African-American police officer from Colorado, successfully manages to infiltrate the local Ku Klux Klan and become the head of the local chapter.',
        vote_average: 7.8,
        poster_path:
          'http://image.tmdb.org/t/p/original/55W6mUVv4CXMMQHHhV2zXtLSpXQ.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 487258,
        title: 'Marainthirunthu Paarkum Marmam Enna',
        release_date: '2018-08-17',
        overview:
          "A daring young man joins a gang that is behind chain-snatching crimes in the city. What's his game plan?",
        vote_average: 5.2,
        poster_path:
          'http://image.tmdb.org/t/p/original/2zfyj4gY7RHkf89ssSTly2UppMC.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 455207,
        title: 'Crazy Rich Asians',
        release_date: '2018-08-15',
        overview:
          "An American-born Chinese economics professor accompanies her boyfriend to Singapore for his best friend's wedding, only to get thrust into the lives of Asia's rich and famous.",
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/gnTqi4nhIi1eesT5uYMmhEPGNih.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 400155,
        title: 'Hotel Transylvania 3: Summer Vacation',
        release_date: '2018-06-30',
        overview:
          "Dracula, Mavis, Johnny and the rest of the Drac Pack take a vacation on a luxury Monster Cruise Ship, where Dracula falls in love with the ship's captain, Ericka, who's secretly a descendant of Abraham Van Helsing, the notorious monster slayer.",
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/gjAFM4xhA5vyLxxKMz38ujlUfDL.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 420814,
        title: 'Christopher Robin',
        release_date: '2018-08-02',
        overview:
          'Working-class family man Christopher Robin encounters his childhood friend Winnie-the-Pooh, who helps him to rediscover the joys of life.',
        vote_average: 7.3,
        poster_path:
          'http://image.tmdb.org/t/p/original/xR5w0he6czZkcAz459a4iPBqXGe.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      }
    ],
    favorites: [402900, 455980, 345887, 345940]
  },
  currentUser: { id: 2, name: 'asdf', password: 'asdf', email: 'asdf' }
};
