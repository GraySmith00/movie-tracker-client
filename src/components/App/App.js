import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  NavLink,
  Switch
} from 'react-router-dom';
import { connect } from 'react-redux';

import { getNowPlaying, populateSearch, getFavorites } from '../../helpers.js';

import { addNowPlaying, updateFavorites } from '../../actions/movieActions';
import Login from '../../containers/Login/Login';
import Register from '../../containers/Register/Register';
import './App.css';
import CardContainer from '../CardContainer/CardContainer.js';
import { setCurrentUser } from '../../actions/userActions.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      activeTab: ''
    };
  }

  async componentDidMount() {
    const nowPlaying = await getNowPlaying();
    this.props.addNowPlaying(nowPlaying);
  }

  handleClick = event => {
    const { name } = event.target;
    this.setState({ activeTab: name });
  };

  logoutUser = () => {
    this.props.setCurrentUser(null);
  };

  setFavoritesState = async () => {
    const { currentUser, updateFavorites } = this.props;
    if (currentUser) {
      const favorites = await getFavorites(currentUser);
      updateFavorites(favorites.data);
    }
  };

  render() {
    return (
      <div className="App">
        <Router>
          <div>
            <header>
              <nav className="nav-btns">
                <NavLink exact to="/" className="nav-link">
                  Home
                </NavLink>
                <NavLink exact to="/login" className="nav-link">
                  Login
                </NavLink>
                <NavLink exact to="/register" className="nav-link">
                  Sign Up
                </NavLink>
                <NavLink
                  exact
                  to="/"
                  className="nav-link"
                  onClick={this.logoutUser}
                >
                  Logout
                </NavLink>
                <NavLink exact to="/favorites" className="nav-link">
                  Favorites
                </NavLink>
              </nav>
            </header>
            <main>
              <h1>App</h1>
              <Switch>
                <Route
                  exact
                  path="/"
                  render={() => {
                    return <CardContainer category={'nowPlaying'} />;
                  }}
                />
                <Route exact path="/login" component={Login} />
                <Route exact path="/register" component={Register} />
                <Route
                  exact
                  path="/favorites"
                  render={() => {
                    this.setFavoritesState();
                    return <CardContainer category={'favorites'} />;
                  }}
                />
              </Switch>
            </main>
          </div>
        </Router>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  updateFavorites: favorites => dispatch(updateFavorites(favorites)),
  addNowPlaying: movies => dispatch(addNowPlaying(movies)),
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
