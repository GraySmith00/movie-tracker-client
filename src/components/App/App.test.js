import React from 'react';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps } from './App';
import { mockNowPlayingFetch } from '../../mockData/mockFetches';
import { mockStore } from '../../mockData/mockStore';
import { addNowPlaying, clearFavorites } from '../../actions/movieActions';

describe('App', () => {
  let wrapper;
  let mockAddNowPlaying;
  let mockSetCurrentUser;
  let mockClearFavorites;

  beforeEach(async () => {
    mockAddNowPlaying = jest.fn();
    mockSetCurrentUser = jest.fn();
    mockClearFavorites = jest.fn();
    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockNowPlayingFetch)
      })
    );
    wrapper = await shallow(
      <App
        addNowPlaying={mockAddNowPlaying}
        setCurrentUser={mockSetCurrentUser}
        clearFavorites={mockClearFavorites}
        currentUser={mockStore.currentUser}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should make a fetch call on compoenent did mount', () => {
    expect(window.fetch).toHaveBeenCalled();
  });

  it.skip('should catch if an error', async () => {
    window.fetch = jest
      .fn()
      .mockImplementation(() => Promise.reject(new Error('failed')));

    const error = new Error('failed');

    wrapper = shallow(
      <App
        addNowPlaying={mockAddNowPlaying}
        setCurrentUser={mockSetCurrentUser}
        clearFavorites={mockClearFavorites}
        currentUser={mockStore.currentUser}
      />
    );
    await expect(wrapper).rejects.toEqual(error);
  });

  describe('mapDispatchToProps', () => {
    let movies = [
      {
        movie_id: 345940,
        title: 'The Meg',
        release_date: '2018-08-09',
        overview:
          'A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.',
        vote_average: 6.2,
        poster_path:
          'http://image.tmdb.org/t/p/original/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg',
        favorite: false
      }
    ];
    it('should map the store correctly if addNowPlaying is dispatched', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      mapped.addNowPlaying(movies);
      expect(mockDispatch).toHaveBeenCalled();
    });
    it('should map the store correctly if setCurrentUser is dispatched', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      mapped.setCurrentUser(mockStore.currentUser);
      expect(mockDispatch).toHaveBeenCalled();
    });
    it('should map the store correctly if clearFavorites is dispatched', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      mapped.clearFavorites();
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});
