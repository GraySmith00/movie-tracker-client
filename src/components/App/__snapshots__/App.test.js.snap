// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addNowPlaying={[Function]}
    clearFavorites={[Function]}
    currentUser={
        Object {
            "email": "asdf",
            "id": 2,
            "name": "asdf",
            "password": "asdf",
          }
    }
    setCurrentUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <BrowserRouter>
        <div>
                <header>
                        <Unknown
                                logoutUser={[Function]}
                        />
                </header>
                <main>
                        <h1>
                                App
                        </h1>
                        <Unknown />
                </main>
        </div>
</BrowserRouter>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <header>
                    <Unknown
                              logoutUser={[Function]}
                    />
          </header>
          <main>
                    <h1>
                              App
                    </h1>
                    <Unknown />
          </main>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header>
              <Unknown
                            logoutUser={[Function]}
              />
</header>,
            <main>
              <h1>
                            App
              </h1>
              <Unknown />
</main>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Unknown
                logoutUser={[Function]}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "logoutUser": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  App
</h1>,
                <Unknown />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "App",
                },
                "ref": null,
                "rendered": "App",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "main",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BrowserRouter>
          <div>
                    <header>
                              <Unknown
                                        logoutUser={[Function]}
                              />
                    </header>
                    <main>
                              <h1>
                                        App
                              </h1>
                              <Unknown />
                    </main>
          </div>
</BrowserRouter>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            <header>
                        <Unknown
                                    logoutUser={[Function]}
                        />
            </header>
            <main>
                        <h1>
                                    App
                        </h1>
                        <Unknown />
            </main>
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <header>
                <Unknown
                                logoutUser={[Function]}
                />
</header>,
              <main>
                <h1>
                                App
                </h1>
                <Unknown />
</main>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Unknown
                  logoutUser={[Function]}
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "logoutUser": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    App
</h1>,
                  <Unknown />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "App",
                  },
                  "ref": null,
                  "rendered": "App",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "main",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
